clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -0,0 +1,279 @@
+package com.linkedin.venice.datarecovery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * StoreRepushCommand contains the details of executing/processing repush command.
+ * We expect the command to comply with the following contract:
+ *
+ * Input:
+ *    <COMMAND> --store <store_name> --fabric <source_fabric> --password <credentials> [<EXTRA_COMMAND_ARGS>]
+ * Output:
+ *    success: link_to_running_task
+ *    failure: failure_reason
+ */
+
+public class StoreRepushCommand {
+  private static final Logger LOGGER = LogManager.getLogger(StoreRepushCommand.class);
+
+  // Store name.
+  private String store;
+  private Params params;
+  private Result result;
+  private List<String> shellCmd;
+
+  public StoreRepushCommand() {
+  }
+
+  public StoreRepushCommand(String store, Params params) {
+    this.store = store;
+    this.params = params;
+    this.shellCmd = generateShellCmd();
+  }
+
+  public void setParams(Params params) {
```

Comment:
Is this `setter` necessary?

Reply:
It is only used in unit test. Will add a comment.